lib/ood_core/job/adapters/systemd/launcher.rb

```diff
@@ -97,27 +99,34 @@ def call(cmd, *args, env: {}, stdin: "")
   # if ! strict_host_checking
   # -o UserKnownHostsFile=/dev/null (do not update the user's known hosts file)
   # -o StrictHostKeyChecking=no (do no check the user's known hosts file)
+  # if ssh_keyfile
+  # -i ssh_keyfile (Use this keyfile location)
   #
   # @param destination_host [#to_s] the destination host you wish to ssh into
   # @param cmd [Array<#to_s>] the command to be executed on the destination host
   def ssh_cmd(destination_host, cmd)
-    if strict_host_checking
-      [
-        'ssh', '-t',
-        '-p', OodCore::Job::Adapters::Helper.ssh_port,
-        '-o', 'BatchMode=yes',
-        "#{username}@#{destination_host}"
-      ].concat(cmd)
-    else
-      [
-        'ssh', '-t',
-        '-p', OodCore::Job::Adapters::Helper.ssh_port,
-        '-o', 'BatchMode=yes',
-        '-o', 'UserKnownHostsFile=/dev/null',
-        '-o', 'StrictHostKeyChecking=no',
-        "#{username}@#{destination_host}"
-      ].concat(cmd)
+
+    basecmd=[
+      'ssh', '-t',
+      '-p', OodCore::Job::Adapters::Helper.ssh_port,
+      '-o', 'Batchmode=yes',
+      "#{username}@#{destination_host}",
+    ].concat(cmd)
+
```

Comment:
I think a structure like this is more idiomatic for Ruby.

```ruby
args = ['ssh', '-t'] # plus more, just for brevity
args.concat([...]) << if strict_host_checking
args.concat([...]) << unless !ssh_keyfile.to_s.empty?

args
```

Reply:
I'll give it a test, I've never written ruby before so it's all new to me 😋

Reply:
I can edit this PR, I'm happy to take care of it if you like.

Reply:
I think I've updated it adequately. If not please feel free to change it up.

Reply:
Thanks, sorry for the delay. I'll get around to this soon.

Reply:
Yes there was, I've now removed this.